[rust] outdated requirements documentation in regards to dependencies

https://github.com/greenbone/openvas-scanner/blob/main/rust/README.md#requirements misses net-snmp (-lnetsnmp) and states that openssl (-lssl" "-lcrypto) is only required for experimental features, which doesn't seem to be true anymore.

`"-Wl,-Bdynamic" "-lnetsnmp" "-lssl" "-lcrypto" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-L"`

For CMake there's the option to disable net-snmp via `-DBUILD_WITH_NETSNMP:BOOL=FALSE` (https://github.com/greenbone/openvas-scanner/blob/main/CMakeLists.txt#L142-L144) but no cargo feature analog to this for the rust code as far as I can see, is this planned in the future or now a mandatory dependency as well?

### Expected behavior
Compile without having net-snmp (and openssl) installed.

### Actual behavior
Fails to build without having net-snmp (and openssl) installed:

```
[...]
_std_workspace_core-*,liballoc-*,libcore-*,libcompiler_builtins-*}.rlib" "-Wl,-Bdynamic" "-lnetsnmp" "-lssl" "-lcrypto" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-L" "/var/tmp/paludis/build/net-analyzer-openvas-scanner-23.38.4/temp/rustczrAddj/raw-dylibs" "-B<sysroot>/lib64/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld" "-fuse-ld=lld" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/var/tmp/paludis/build/net-analyzer-openvas-scanner-23.38.4/work/openvas-scanner-23.38.4/rust/target/release/build/aws-lc-sys-0f030b5864836398/out" "-L" "/var/tmp/paludis/build/net-analyzer-openvas-scanner-23.38.4/work/openvas-scanner-23.38.4/rust/target/release/build/ring-96136f6a36093d6a/out" "-L" "/var/tmp/paludis/build/net-analyzer-openvas-scanner-23.38.4/work/openvas-scanner-23.38.4/rust/target/release/build/libsqlite3-sys-06c1ce24a83b1af9/out" "-L" "/var/tmp/paludis/build/net-analyzer-openvas-scanner-23.38.4/work/openvas-scanner-23.38.4/rust/target/release/build/zstd-sys-0112174632fada4d/out" "-L" "<sysroot>/lib/openssl-3/lib" "-L" "<sysroot>/lib64/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/var/tmp/paludis/build/net-analyzer-openvas-scanner-23.38.4/work/openvas-scanner-23.38.4/rust/target/release/deps/openvasd-e9b734ad23cb7182" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,-O1" "-Wl,--strip-debug" "-nodefaultlibs"
  = note: some arguments are omitted. use `--verbose` to show all linker arguments
  = note: ld.lld: error: unable to find library -lnetsnmp
          collect2: error: ld returned 1 exit status
          

error: could not compile `scannerlib` (bin "openvasd") due to 1 previous error
```

### Steps to reproduce

1. do not have net-snmp installed
2. try to build the rust code `cargo build --release`

### GVM versions

**gsa:** -

**gvm:** -

**openvas:** -

**gvm-libs:** -

**openvas-smb:** -

**ospd-openvas:** -

### Environment

**Operating system:** Exherbo Linux

**Installation method / source:** source based distribution / packages

### Logfiles
[openvas-scanner-23.38.4_build.log](https://github.com/user-attachments/files/25051476/openvas-scanner-23.38.4_build.log)

--- a/rust/README.md
+++ b/rust/README.md
@@ -19,11 +19,12 @@ The implementation is split into multiple parts that are reflected in the direct
 # Requirements
 
 - rust toolchain
+- openssl
+- libnetsnmp
 
 Additionally for the features defined as experimental you need:
 
 - libpcap
-- openssl
 - pkg-config
 - zlib
 

